lib/api/src/sys/externals/function.rs

```diff
@@ -66,6 +66,44 @@ impl wasmer_types::WasmValueType for Function {
     }
 }
 
+impl Function {
+    /// Convert a `VMFuncRef` into a `Function`.
+    ///
+    /// Returns `None` if the funcref is null.
+    pub fn from_vm_funcref(store: &Store, func_ref: VMFuncRef) -> Option<Self> {
```

Comment:
`VMCallerCheckedAnyFunc` itself just a reference to actual machine code, but the point still stands – nothing here prevents `Function` from outliving the function.

> (Also, I'm surprised that the one in types.rs was not unsafe already, but I guess it's because it was not publicly exposed before)

The functionality is largely public through `Val` already, so yeah…